server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectPolicyOperations.java

```diff
@@ -0,0 +1,348 @@
+              }
+
+              Policy[] parentPolicies =
+                  policyDispatcher.listPolicyInfosForMetadataObject(metalake, parentObject);
+              if (hasConflictExclusivePolicy(parentPolicies, targetPolicy)) {
+                // If another same type exclusive policy is found, it means the target policy will
+                // be replaced by the same type policy
+                return logNotFoundPolicy(metalake, policyName, type, fullName);
+              }
+
+              Optional<Policy> inheritedPolicy =
+                  ArrayUtils.isEmpty(parentPolicies)
+                      ? Optional.empty()
+                      : Arrays.stream(parentPolicies)
+                          .filter(p -> p.name().equals(policyName))
+                          .findFirst();
+              if (inheritedPolicy.isPresent()) {
+                // If the policy is found in parent object, we convert it to DTO and return.
+                LOG.info(
+                    "Found policy: {} for object type: {}, full name: {} under metalake: {}",
+                    policyName,
+                    type,
+                    fullName,
+                    metalake);
+                return Utils.ok(
+                    new PolicyResponse(
+                        DTOConverters.toDTO(inheritedPolicy.get(), Optional.of(true))));
+              }
+
+              parentObject = MetadataObjects.parent(parentObject);
+            }
+
+            return logNotFoundPolicy(metalake, policyName, type, fullName);
+          });
+
+    } catch (Exception e) {
+      return ExceptionHandlers.handlePolicyException(OperationType.GET, policyName, fullName, e);
+    }
+  }
+
+  @GET
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "list-object-policies." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "list-object-policies", absolute = true)
+  public Response listPoliciesForMetadataObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      @QueryParam("details") @DefaultValue("false") boolean verbose) {
+    LOG.info(
+        "Received list policy {} request for object type: {}, full name: {} under metalake: {}",
+        verbose ? "infos" : "names",
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+
+            Set<PolicyDTO> policies = new HashSet<>();
+            Set<String> exclusivePolicyTypes = new HashSet<>();
+            Policy[] nonInheritedPolicies =
+                policyDispatcher.listPolicyInfosForMetadataObject(metalake, object);
+            for (Policy policy : nonInheritedPolicies) {
+              if (policy.exclusive()) {
```

Comment:
Do we check the property `exclusive`  in the association operation? Why do we need to check it again?

Reply:
it's used to calculate the inherited value